### ISSUE TITLE
Remove the implementation section, shove it all in the header

### ISSUE THREAD

**Issue by bvisness** [HandmadeMath/Handmade-Math#116]
I know we've gone back and forth on this before (#57) but I don't think we need to have a separate implementation section in Handmade Math.

Most of the function implementations are inline, since that gives the compiler the most context for optimizations (and math needs to be optimized!). The ones that are _not_ inline were made that way because we were worried about redundant definitions between translation units, but with link-time optimization, compilers are mostly able to prevent that from happening.

So basically we are forcing the user to set up the `HANDMADE_MATH_IMPLEMENTATION` part for about three functions, and for little to no benefit.

I propose that we make everything in Handmade Math inline for 2.0, and rip out `HANDMADE_MATH_IMPLEMENTATION` completely.

**Comment by StrangeZak** [comment]
I’m fully onboard for this. LETS DO IT 👍
